docs/products/kafka/concepts/acl.md

```diff
@@ -1,116 +1,216 @@
+- **Associated topics**: Specific Kafka topics or wildcard patterns.
 
-Examples:
+Aiven checks ACL entries and grants access if there's a match. If no entry matches,
+access is denied. The order of ACL entries does not
+influence access evaluation.
 
--   username: `abc`, permission: `read`, topic: `xyz`. User `abc` has
-    read access to topic `xyz`.
--   username: `analyst*`, permission: `read`, topic: `xyz`. All Aiven
-    users with username starting `analyst` have read access to topic
-    `xyz`.
--   username: `developer*`, permission: `read`, topic: `test*`. All
-    Aiven users with username starting `developer` have read access to
-    topics starting with `test`.
+### Examples
 
-:::warning
-By default, Aiven adds an `avnadmin` service user to every new service
-and adds `admin` permission for all topics to that user. When you create
-your own ACLs to restrict access, you probably want to remove this ACL
-entry.
-:::
+- User-specific access:
+
+  ```plaintext
+  username: abc
+  permission: read
+  topic: xyz
+  ```
+
+  Grants user `abc` read access to the topic `xyz`.
+
+- Wildcard-based access:
+
+  ```plaintext
+  username: analyst*
+  permission: read
+  topic: xyz
+  ```
+
+  Grants all users with usernames starting with `analyst` read access to the topic `xyz`.
+
+- Wildcard in topics:
+
+  ```plaintext
+  username: developer*
+  permission: read
+  topic: test*
+  ```
+
+  Grants all users with usernames starting with `developer` read access to topics
+  starting with `test`.
+
+## Kafka-native ACL capabilities
+
+Kafka-native ACLs provide granular, resource-level permissions for precise access
+control. They enable the management of Apache Kafka topics and additional resources
+like clusters, consumer groups, and transactional IDs. It is ideal for
+advanced scenarios requiring fine-tuned access management.
+
+- **Fine-grained permissions**: Allow both `ALLOW` and `DENY` rules to provide
+  precise control over access.
+- **Expanded resource-level control**: Manage access to non-topic resources,
+  such as consumer groups, clusters, and transactional IDs.
+- **Pattern-based matching**: Specify how resource names are matched using `LITERAL`
+  for exact matches or `PREFIXED` for prefixes.
+
+### ACL structure
+
+A Kafka-native ACL entry consists of the following elements:
+
+- **Principal**: Specifies the user or service account, such as `User:Alice`.
+- **Host**: Defines the allowed host. Use `*` to allow all hosts.
+- **Resource type**: Indicates the Apache Kafka resource to control, such as `Topic`,
+  `Group`, `Cluster`, or `TransactionalId`.
+- **Pattern type**: Determines how the resource value is matched:
+  - **LITERAL**: Matches an exact resource name, such as `my-topic`.
+  - **PREFIXED**: Matches all resources sharing a specified prefix, such as `logs-*`.
+- **Resource**: Refers to the specific Kafka resource. Its value depends on the selected pattern type.
```

Comment:
There is also pattern type = LITERAL + resource = `*` which as s special case matches everything. That should be documented somewhere. It is standard Kafka convention though.

Reply:
This a note to clarify this.